### DISCUSSION TITLE
libiperf.so versioning

### DISCUSSION THREAD

**Discussion by allmondjoy87** [esnet/iperf#1638] [Q&A] [answered]
I'm compiling iperf to use libiperf and the c api, but whenever I build the library it's built as `libiperf.so.0.0.0`. I've tried this with the master and 3.16 tagged release and gotten the same results. I'm not using any funky `./configure` parameters, don't get any related errors during configure or make, and the resulting binaries/libs all work without error.

Any ideas what I'm missing here? Thanks.

**Comment by bmah888** [comment, marked as answer]
I think that's just the default shared library version. Is there something wrong with that number?

Bruce.

**Reply by allmondjoy87** [reply]
No problem at all! I just anticipated it would produce something like this for version control:
```
libiperf.so -> libiperf.so.3.16.0
libiperf.so.3 -> libiperf.so.3.16.0
libiperf.so.3.16.0
```
Glad to know the `0.0.0` is expected behavior and not an issue with my build. Thanks.
